gemrb/core/Game.cpp

```diff
@@ -1568,10 +1568,10 @@ void Game::UpdateScripts()
 		doChangeSongHard = false;
 	} else if (!core->GetMusicMgr()->IsPlaying()) {
 		// perhaps a StartMusic action stopped the area music?
-		// (we should probably find a less silly way to handle this,
-		// because nothing can ever stop area music now..)
-		doChangeSong = true;
+		doChangeSong = InterruptedMusic;
 		doChangeSongHard = false;
+
+		InterruptedMusic = false;
```

Comment:
This whole commit looks suspicious as well. Do you have any test case or did you just go from the ancient comment?

Reply:
I actually don't have a test case. I started from that comment and looked when an area song change can happen, i.e. when is PlayAreaSong called: change maps, enter/leave combat, and StartMusic action. Setting a flag in StartMusic to signal this and using this flag here in order to prevent ChangeSong spams for no-music-areas seems sensible to me.

Reply:
This out of band communication is ugly and a waste of a member variable. In what case would StartMusic get into a conflict here? PlayAreaSong will start the requested music, so IsPlaying will be true anyway.

Btw, StartMusic has only 3 uses in the originals, for boss music.

Reply:
Ah ok, you are right, when StartMusic starts a new music it can only be stopped by either map change or combat change. But that also means that the comment there is obsolete, and it begs the question if the IsPlaying test and the associated ChangeSong call here do actual have a reason to exist? It can function as a failsafe for unexpected stopped music at most.

So I will drop this commit. But we need to stop the ChangeSong spam with some other measurement.

Reply:
What do you mean with ChangeSong spam?

Reply:
The !IsPlaying test and the associated ChangeSong call are executed on every UpdateScripts call in an area without music.

Reply:
An easy mitigation would be to only check it once per round.

Reply:
Thanks, done.